src/transaction.rs

```diff
@@ -572,6 +575,48 @@ impl TxOut {
         }
     }
 
+    /// Convert into an explicit output if possible.
+    pub fn into_explicit(self) -> Option<ExplicitTxOut> {
+        Some(ExplicitTxOut {
+            asset: self.asset.explicit()?,
+            value: self.value.explicit()?,
+            script_pubkey: self.script_pubkey,
+            nonce: self.nonce,
+        })
+    }
+
+    /// Convert into an explicit output if possible.
+    pub fn to_explicit(&self) -> Option<ExplicitTxOut> {
+        Some(ExplicitTxOut {
+            asset: self.asset.explicit()?,
+            value: self.value.explicit()?,
+            script_pubkey: self.script_pubkey.clone(),
+            nonce: self.nonce,
+        })
+    }
+
+    /// Convert into a confidential output if possible.
+    pub fn into_confidential(self) -> Option<ConfidentialTxOut> {
+        Some(ConfidentialTxOut {
+            asset: self.asset.commitment()?,
+            value: self.value.commitment()?,
+            nonce: self.nonce,
+            script_pubkey: self.script_pubkey,
+            witness: self.witness,
+        })
+    }
+
+    /// Convert into a confidential output if possible.
+    pub fn to_confidential(&self) -> Option<ConfidentialTxOut> {
+        Some(ConfidentialTxOut {
+            asset: self.asset.commitment()?,
+            value: self.value.commitment()?,
+            nonce: self.nonce,
+            script_pubkey: self.script_pubkey.clone(),
+            witness: self.witness.clone(),
+        })
+    }
```

Comment:
In case there is controversy about these, we could also add the `unblind` API directly on `TxOut` and simply extend the `Error` type with a variant `NotConfidential`.

Reply:
Yeah, I would prefer this approach. In principle you can have txouts that are "half blinded" (explicit asset confidential value, or vice versa) and I'd like to avoid excluding these in the type system.

Reply:
@apoelstra I've adapted the PR based on this thread and the changes that have been introduced since I opened it.